File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet <Integer> set = new HashSet <>();
        int longest = 0;
        for (int i : nums){
            set.add(i);
        }
        for (int i : set){
            if(!set.contains(i - 1)){
                int curr = i ;
                int currLength = 1;
                while (set.contains (curr + 1)){
                    curr++;
                    currLength ++;
                }
                longest = Math.max(longest, currLength);
            }
        }
        return longest;
    }
}

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int length = 0;
        unordered_set <int> s;
        for(int i : nums){
            s.insert(i);
        }
        for(int i : s){
            if(s.count(i - 1) == 0){
                int m = i + 1;
                while(s.count(m) != 0){
                    m++;
                }
                length = max(length, m - i);
            }
        }
        return length;
    }
};